source/extensions/common/wasm/context.cc

```diff
@@ -427,8 +441,13 @@ WasmResult Context::getProperty(absl::string_view path, std::string* result) {
         value = CelValue::CreateMessage(&info->dynamicMetadata(), &arena);
         break;
       case PropertyToken::FILTER_STATE:
-        value = CelValue::CreateMap(
-            Protobuf::Arena::Create<WasmStateWrapper>(&arena, info->filterState()));
+        if (getConnection()) {
```

Comment:
Store the connection in a local and check if it is nullptr then reuse that local to get the filter state.